### PR TITLE
A small extraction and some removed ifs.

### PR DESCRIPTION
**Pull request by Skarlso**
Added test for issue #3.

### CODE REVIEW

File: lib/jsonpathv2/enumerable.rb [hunk 1]
```diff
@@ -64,9 +63,7 @@ def handle_wildecard(node, expr, context, key, pos, &blk)
             next unless start_idx
             end_idx = (array_args[1] && process_function_or_literal(array_args[1], -1) || (sub_path.count(':') == 0 ? start_idx : -1))
             next unless end_idx
-            if start_idx == end_idx
-              next unless start_idx < node.size
-            end
+            next if start_idx == end_idx && start_idx > node.size
```
Comment (cantino) [inline lib/jsonpathv2/enumerable.rb:66]: Shouldn't this be `>= node.size`?

Comment (Skarlso) [reply lib/jsonpathv2/enumerable.rb:66]: You know what @cantino.. The tests are passing either way. :/ Damn. I have this feeling, that I also need to re-write some of the tests. I'm not a 100% sure if they truly cover what they say they cover. :)

Comment (cantino) [reply lib/jsonpathv2/enumerable.rb:66]: I hate when I find holes in specs :)

Comment (Skarlso) [reply lib/jsonpathv2/enumerable.rb:66]: Yeah, I know the feeling. :) Especially when taking over code from somewhere. :D And you sort of rely on the tests to guide you and be there for you while you slowly explore just to find out that they are lying. :D 

lied

Comment (cantino) [reply lib/jsonpathv2/enumerable.rb:66]: 😢

**Pull request merged**